modules/reindex/src/main/java/org/elasticsearch/reindex/ReindexValidator.java

```diff
@@ -160,6 +160,11 @@ private static SearchRequest skipRemoteIndexNames(SearchRequest source) {
     }
 
     private static boolean isRemoteExpression(String expression) {
-        return expression.contains(":");
+        // remote index name contains ':' symbol inside their names (remote cluster name separator), e.g. cluster0:index-name
```

Comment:
This comment is worded slightly incorrectly. I think the better way to say it would be:

> An index expression that references a remote cluster uses ":" to separate the cluster-alias from the index portion of the expression, e.g., cluster0:index-name ...